scenario_player/setup/__init__.py

```diff
@@ -0,0 +1,157 @@
+"""Scenario Player Setup Utilities.
+
+In order for the scenario player to run a scenario, it must first setup
+certain components, in order to be able to execute any assertions stated in a
+scenario yaml file.
+
+The following paragraphs will briefly describe the steps that are taken along
+the setup phase of a Scenario Player run.
+
+For details, please read the documentation of the respective modules in
+:mod:`scenario_player.setup`.
+
+
+..note::
+
+    Before we continue, we'd like to ensure that some of the less-well defined
+    terminology. The following is a list of terms which are used throughout the
+    raiden code base, but tend to mean/talk about different things. To ensure
+    we do not get lost in translation, here are their definitions as understood
+    in this repository:
+
+        Account
+        Keystore
+        Wallet
+            The file holding the ethereum address, which in turn holds funds.
+
+        SP Keystore
+        SP Account
+            The Keystore file which holds the funds for a scenario. All Raiden
+            nodes created for this scenario are funded using this account's ether.
+            These funds also pay for Token Contract deployments and RPC calls
+            during setup.
+
+        Node Keystore
+        Node Account
+            The keystore file explicitly created for a node. These are always
+            password protected by an empty string and automatically generated
+            if they do not exist during setup. Needless to say, it's a terrible
+            idea to store actual ether on these.
+
+        Node
+        Raiden Client
+        Raiden Nodes
+            Instances of a `raiden` binary, started with a specific set of
+            CLI options.
+
+        SP Token Contract
+            The custom token contract deployed or loaded during a scenario run.
+            It mints SP Tokens, which are sent between raiden client nodes to
+            assert their behaviour's correctness.
+
+
+Parse and Validate the scenario YAML
+====================================
+
+The scenario file is read into memory using :mod:`pyyaml` and loaded using the
+tools provided in :mod:`scenario_player.utils.settings`. The latter also validate
+the values given for compatibility and correctness, as well as filling in
+defaults for absent optional values.
+
+
+Create Raiden Nodes
+===================
+
+Create Keystores
+----------------
+
+In this step, we create keystores programmatically for each required Node, as long as
+a keystore does not exist already for it.
+
+Construct CLI flags
+-------------------
+This step constructs the list of command-line flags to pass to the binary. First,
+the default/general options are created, and then overriden per node as requested
+in the scenario yaml.
+
+Spawn Client in subprocess
+--------------------------
+Spawn all requested nodes using :mod:`subprocess`. There is no further direct
+interaction between the SP and each node's binary after this point.
```

Comment:
I steered clear of mentioning mirakuru, since a) this was supposed be brief, and b) I wanted the overview to be fairly free of implementation detail. I'll make sure to include that in the `scenario_player.setup.nodes` docs though, of course.

Reply:
Just noticed I didnt steer clear when mentioning the loading of the scenario file - will put my money where my mouth is and fix that too.